<?xml version="1.0"?>
<launch>
  <!-- Parameters to pass to the ROS node -->
  <arg name="move_group" default="manipulator"/>
  <arg name="ee_link" default="tool_link"/>
  <arg name="ros_rate" default="100"/>
  <arg name="model" default="iiwa7_tool"/>
  <arg name="sim" default="true"/>
  <arg name="world" default="iiwa"/>
  <arg name="tool_name" default="tool" />

  <param name = "/iiwa/toolName" type="string" value="$(arg tool_name)" />

  <!--  This loads the robot description  -->
  <include file="$(find iiwa_examples)/launch/moveit_planning_example.launch">
    <arg name="sim" value="$(arg sim)"/>
    <arg name="world" value="$(arg world)"/>
    <arg name="model" value="$(arg model)"/>
  </include>
    
  <!-- Here we call the ROS node we have written, with the parameters on top   -->
  <node ns="iiwa" name="CommandRobotIIWA" pkg="iiwa_examples" type="drawing_spline" respawn="false" output="screen">
    <param name="move_group" value="$(arg move_group)"/>
    <param name="ee_link" value="$(arg ee_link)"/>
    <param name="ros_rate" value="$(arg ros_rate)"/>
  </node>
  
</launch>
